### ISSUE TITLE
Alligning with Martin Günther

### ISSUE THREAD

**Issue by kdhansen** [relffok/mir_robot#32]
It seems you are doing good work on porting these packages. I saw that @mintar (Martin Günther) is also working on a ros2 branch. He has been merging some of your commits, but have also contributed a good deal himself. Are you aware of his work, could it make sense to collaborate?

**Comment by mintar** [comment]
Yes, I am very aware of @relffok's great work on porting the mir_driver repository to ROS2. She sent me a pull request (DFKI-NI/mir_robot#96) **two years** ago, and I **still** have not managed to finish working on it (sorry!). I have already spent many days on it, but it's still not finished; I really hope I will get around to finishing it soon.

My current work is here: https://github.com/mintar/mir_robot/tree/ros2

... but I already have more local changes that I haven't yet pushed to that branch. My plan there is to integrate @relffok's commits and fix/add some stuff that went missing during the conversion, and afterwards port over all the work from `noetic` to `ros2` that has been committed to `noetic` since the two branches diverged. The goal is to keep the `ros2` and `noetic` branches as similar as possible, so it's easier in the future to port over changes. For example, there is no need to use the Python launch file syntax in ROS2; nowadays, there's a XML frontend for ROS2 launch files that looks very similar to the ROS1 launch file syntax. I plan to use the XML format everywhere so that future PRs to Noetic can be easily adapted to ROS2 and vice versa.

It's still a lot of work, that's why I haven't finished doing it yet.

**Comment by kdhansen** [comment]
Thanks for your work Martin!
I have forked your repo and cherrypicked most of @relffok 's commits on her branch here:

https://github.com/AAU-RoboticsAutomationGroup/mir_robot

But it sounds like you have a lot local changes and plans, so I don't know if the work in my fork can help.

**Comment by dayvsonsilva** [comment]
Hello! Thank you very much for your work on this project! I'm testing the repository that @kdhansen  presented on a physical robot, I'm having some difficulties and would like to know if it's possible to enable the issuis tab in the repository [AAU- RoboticsAutomationGroup/mir_robot](https://github.com/AAU-RoboticsAutomationGroup/mir_robot) so that doubts are concentrated in the test repository.

**Comment by kdhansen** [comment]
> I'm having some difficulties and would like to know if it's possible to enable the issuis tab in the repository [AAU- RoboticsAutomationGroup/mir_robot](https://github.com/AAU-RoboticsAutomationGroup/mir_robot) so that doubts are concentrated in the test repository.

There you go :)

**Comment by shrutic12** [comment]
Hello @kdhansen @mintar ! Have there been any new updates on the ROS2 support? Is there currently a version that can be tested with the real robot?

**Comment by mintar** [comment]
We are actively working on it! Right now, the most current version is at https://github.com/marcvinci/mir_robot/tree/humble-devel (don't be confused by the branch name `humble-devel`, the target distribution is ROS2 Jazzy). It's being tested on the real robot, and most things are already working.

This is only a temporary fork though! Once everything's stable, we will merge it here.

**Comment by relffok** [comment]
Hello @kdhansen, very sorry for the late reply. I am aware of @mintar's work. After all, this is a fork of his ROS1 repository. Happy to see that you could re-use some of my work! As you might have noticed, there was no further development for quite a while now. Mostly since I have no access to a MiR robot anymore.. Looking forward to seeing your final version be merged!
